functional/verifier_registrar_unreachable/test.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+AGENT_ID="d432fbb3-d2f1-4a97-9ef7-75bd81c00001"
+
+TENANT_ARGS=""
+[ "${AGENT_SERVICE}" == "PushAgent" ] && TENANT_ARGS="--push-model"
+
+rlJournalStart
+
+    rlPhaseStartSetup "Do the keylime setup"
+        [ "${AGENT_SERVICE}" != "Agent" ] && [ "${AGENT_SERVICE}" != "PushAgent" ] && rlDie "Error: AGENT_SERVICE variable is not set. Value 'Agent' or 'PushAgent' expected!"
+        rlRun 'rlImport "./test-helpers"' || rlDie "Error: Cannot import keylime-tests/test-helpers library"
+        rlAssertRpm keylime
+        # update /etc/keylime.conf
+        limeBackupConfig
+        # disable revocation notifications on verifier
+        rlRun "limeUpdateConf revocations enabled_revocation_notifications '[]'"
+        # disable revocation notifications on agent
+        rlRun "limeUpdateConf agent enable_revocation_notifications false"
+        rlRun "limeUpdateConf agent uuid \\\"${AGENT_ID}\\\""
+        # tenant
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+        rlRun "limeUpdateConf verifier exponential_backoff False"
+        rlRun "limeUpdateConf verifier quote_interval 10"
+        rlRun "limeUpdateConf verifier max_retries 50"
+        rlRun "limeUpdateConf verifier request_timeout 20"
+        # configure push attestation
+        if [ "${AGENT_SERVICE}" == "PushAgent" ]; then
+            # Set the verifier to run in PUSH mode
+            rlRun "limeUpdateConf verifier mode 'push'"
+            rlRun "limeUpdateConf verifier challenge_lifetime 1800"
+            rlRun "limeUpdateConf agent attestation_interval_seconds 10"
+            # Right now just for push agent
+            rlRun "limeUpdateConf agent exponential_backoff_initial_delay 1000"
+        fi
+        # if TPM emulator is present
+        if limeTPMEmulated; then
+            # start tpm emulator
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            rlRun "limeCondStartAbrmd"
+            # start ima emulator
+            rlRun "limeInstallIMAConfig"
+            rlRun "limeStartIMAEmulator"
+        fi
+        sleep 5
+        # Need to start verifier for generating
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier"
+        rlRun "limeStopVerifier"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Try to star just agent and check behavior when other services are down"
+        rlRun "limeStart${AGENT_SERVICE}"
+        rlRun -s "limeWaitForAgentRegistration ${AGENT_ID}" 1
+        rlAssertGrep "ERROR - Agent $AGENT_ID does not exist on Registrar" $rlRun_LOG -E
+        #check agent status in logs
+        rlAssertGrep "Error.*Connection refused" $(limeAgentLogfile)
+    rlPhaseEnd
+
+    rlPhaseStartTest "Try register agent after startup registrar, without running verifier, should succeed"
+        limeTIMEOUT=80
```

Comment:
Nové this directly to rlRun where you are waiting and you will not need to restore the value